commons/com.b2international.index/src/com/b2international/index/revision/StagingArea.java

```diff
@@ -733,7 +733,10 @@ public StagingArea stageNew(String key, Object newDocument, boolean commit) {
 		if (stagedObjects.containsKey(objectId)) {
 			StagedObject currentStagedObject = stagedObjects.get(objectId);
 			if (!currentStagedObject.isCommit() && currentStagedObject.getObject() instanceof Revision && newDocument instanceof Revision) {
-				stagedObjects.put(objectId, changed(newDocument, new RevisionDiff((Revision) currentStagedObject.getObject(), (Revision) newDocument), commit));
+				RevisionDiff diff = new RevisionDiff((Revision) currentStagedObject.getObject(), (Revision) newDocument);
+				if (diff.hasChanges()) {
+					stagedObjects.put(objectId, changed(newDocument, diff, commit));
```

Comment:
I think we can create a very minimal test case for this in the index layer module.
1. Use the RevisionData fixture class
2. Start a commit by creating a staging area
3. Add an object to the staging area with commit=false
4. Then add an object to the staging area with commit=true with no changes (assert after this that the staging area did not change)
5. Then add an object to the staging area with commit=true with changes (assert after this that the staging area did change)